File: GestionFactura-Backend/GestionFactura.Domain/Entities/CompanyConfig.cs
namespace GestionFactura.Domain.Entities;

public class CompanyConfig
{
    public int Id { get; set; }
    public string CompanyName { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    
    // config de la base para calculos
    public decimal TaxPercentage { get; set; }
    public string CurrencySymbol { get; set; } = string.Empty;
    
    public string Address { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Region { get; set; } = string.Empty;
    public string PostalCode { get; set; } = string.Empty;
}

File: GestionFactura-Frontend/src/app/core/services/company-config.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

export interface CompanyConfig {
  companyName: string;
  phone: string;
  email: string;
  taxPercentage: number;
  currencySymbol: string;
  address: string;
  city: string;
  region: string;
  postalCode: string;
}

@Injectable({
  providedIn: 'root'
})
export class CompanyConfigService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/companyconfig`;

  getConfig(): Observable<CompanyConfig> {
    return this.http.get<CompanyConfig>(this.apiUrl);
  }

  updateConfig(config: CompanyConfig): Observable<any> {
    return this.http.put(this.apiUrl, config);
  }
}
